jrnl/DayOneJournal.py

```diff
@@ -179,87 +178,45 @@ def write(self):
+                entry.modified = True
+                setattr(entry, attr, new_attr)
+
+    def _get_and_remove_uuid_from_entry(self, entry):
+        uuid_regex = "^ *?# ([a-zA-Z0-9]+) *?$"
+        m = re.search(uuid_regex, entry.body, re.MULTILINE)
+        entry.uuid = m.group(1) if m else None
+
+        # remove the uuid from the body
+        entry.body = re.sub(uuid_regex, "", entry.body, flags=re.MULTILINE, count=1)
+        entry.body = entry.body.rstrip()
+
+        return entry
 
     def parse_editable_str(self, edited):
         """Parses the output of self.editable_str and updates its entries."""
         # Method: create a new list of entries from the edited text, then match
         # UUIDs of the new entries against self.entries, updating the entries
         # if the edited entries differ, and deleting entries from self.entries
         # if they don't show up in the edited entries anymore.
+        entries_from_editor = self._parse(edited)
 
-        # Initialise our current entry
-        entries = []
-        current_entry = None
-
-        for line in edited.splitlines():
-            # try to parse line as UUID => new entry begins
-            line = line.rstrip()
-            m = re.match("# *([a-f0-9]+) *$", line.lower())
-            if m:
-                if current_entry:
-                    entries.append(current_entry)
-                current_entry = Entry.Entry(self)
-                current_entry.modified = False
-                current_entry.uuid = m.group(1).lower()
-            else:
-                date_blob_re = re.compile("^\\[[^\\]]+\\] ")
-                date_blob = date_blob_re.findall(line)
-                if date_blob:
-                    date_blob = date_blob[0]
-                    new_date = jrnl_time.parse(date_blob.strip(" []"))
-                    if line.endswith("*"):
-                        current_entry.starred = True
-                        line = line[:-1]
-                    current_entry.title = line[len(date_blob) - 1 :].strip()
-                    current_entry.date = new_date
-                elif current_entry:
-                    current_entry.body += line + "\n"
-
-        # Append last entry
-        if current_entry:
-            entries.append(current_entry)
+        for entry in entries_from_editor:
+            entry = self._get_and_remove_uuid_from_entry(entry)
 
-        # Now, update our current entries if they changed
-        for entry in entries:
-            entry._parse_text()
-            matched_entries = [
-                e for e in self.entries if e.uuid.lower() == entry.uuid.lower()
-            ]
-            # tags in entry body
-            if matched_entries:
-                # This entry is an existing entry
-                match = matched_entries[0]
-
-                # merge existing tags with tags pulled from the entry body
-                entry.tags = list(set(entry.tags + match.tags))
-
-                # extended Dayone metadata
```

Comment:
How is this extended metadata being preserved? or is it being dropped when the entry is edited?

Reply:
It is being preserved: it is read when the journal is opened, it remains in memory as the entry is being edited, then it's saved back. And it still creates the metadata when writing a new entry.